native/src/CMakeLists.txt

```diff
@@ -75,7 +75,15 @@ target_include_directories(rapidsml_jni PRIVATE
                                     "${cudf_SOURCE_DIR}/java/src/main/native/include"
                                     )
 
-target_link_libraries(rapidsml_jni PRIVATE CUDA::cusolver CUDA::cudart CUDA::cublas cudf::cudf)
+target_link_libraries(rapidsml_jni PRIVATE
+                      libcudart_static.a
```

Comment:
You can use the static versions of the cmake targets, `CUDA::cudart_static` etc. See https://cmake.org/cmake/help/latest/module/FindCUDAToolkit.html

Reply:
Thanks for this shortcut! I tested the target suite locally and met some problems:
The only working writing for this is:
```
target_link_libraries(rapidsml_jni PRIVATE
                      CUDA::cudart_static
                      CUDA::cusparse_static
                      libcusolver_static.a
                      CUDA::cublas_static
                      CUDA::cublasLt_static
                      cudf::cudf
                      libculibos.a
                      liblapack_static.a)
```
`libcusolver` and `libculibos` cannot be replaced. 
But this will result in a jar of 150M(librapidsml_jni.so 284M)
And If I replace `libcusolver_static.a` to `CUDA::cusolver_static` or replace `libculibos.a` to `CUDA::culibos`, some error will be thrown: 
```
undefined symbol: dstedc_gpu
```
or 
```
undefined symbol: culibosInit
```

Reply:
pasted the original successfully compiling command:
```
&& /usr/bin/c++ -fPIC    -shared -Wl,-soname,librapidsml_jni.so -o lib/librapidsml_jni.so src/CMakeFiles/rapidsml_jni.dir/rapidsml_jni.cpp.o src/CMakeFiles/rapidsml_jni.dir/rapidsml_jni.cu.o -L/usr/local/cuda/targets/x86_64-linux/lib/stubs   -L/usr/local/cuda/targets/x86_64-linux/lib 
-Wl,-rpath,/usr/local/cuda/lib64:/home/allxu/miniconda3/lib  
/home/allxu/miniconda3/lib/libcudf.so 
 -Wl,-Bstatic  
-llapack_static  -Wl,-Bdynamic  
/usr/local/cuda/lib64/libcusparse.so  
/usr/local/cuda/lib64/libcusolver_static.a  
/usr/local/cuda/lib64/libcublas_static.a 
 /usr/local/cuda/lib64/libcusparse_static.a  
/usr/local/cuda/lib64/libculibos.a 
 /usr/local/cuda/lib64/libcublasLt.so 
 /home/allxu/miniconda3/lib/libarrow.so.500.0.0  
/home/allxu/miniconda3/lib/libarrow_cuda.so.500.0.0  
/usr/local/cuda/lib64/libcudart.so  
-ldl  -lpthread  /usr/lib/x86_64-linux-gnu/libcuda.so  
-lcudadevrt  
-lcudart_static  
-lrt  -lpthread  -ldl && :
```

And when using: 

```suggestion
                      target_link_libraries(rapidsml_jni PRIVATE
                      CUDA::cudart_static
                      CUDA::cusparse_static
                      CUDA::cusolver_static
                      CUDA::culibos
                      liblapack_static.a
                      CUDA::cublas
                      cudf::cudf)
```
the compile command:
```
&& /usr/bin/c++ -fPIC    -shared -Wl,-soname,librapidsml_jni.so -o lib/librapidsml_jni.so src/CMakeFiles/rapidsml_jni.dir/rapidsml_jni.cpp.o src/CMakeFiles/rapidsml_jni.dir/rapidsml_jni.cu.o -L/usr/local/cuda/targets/x86_64-linux/lib/stubs   -L/usr/local/cuda/targets/x86_64-linux/lib -Wl,-rpath,/usr/local/cuda/lib64:/home/allxu/miniconda3/lib  
-Wl,-Bstatic  -llapack_static  -Wl,-Bdynamic  
/home/allxu/miniconda3/lib/libcudf.so  
/usr/local/cuda/lib64/libcudart_static.a  
/usr/lib/x86_64-linux-gnu/librt.so  
/usr/local/cuda/lib64/libcusparse_static.a  
/usr/local/cuda/lib64/libcusolver_static.a  
/usr/local/cuda/lib64/libculibos.a  
/usr/local/cuda/lib64/libcublas_static.a  
/usr/local/cuda/lib64/libcublas.so  
/home/allxu/miniconda3/lib/libarrow.so.500.0.0  
/home/allxu/miniconda3/lib/libarrow_cuda.so.500.0.0 
 -ldl  -lpthread  /usr/local/cuda/lib64/libcudart.so  
/usr/lib/x86_64-linux-gnu/libcuda.so  
-lcudadevrt  
-lcudart_static  
-lrt  -lpthread  -ldl && :
```

Reply:
I don't understand why cmake put some static libraries(except lapack_static) are put behind `-Bdynamic` along with those shared libraries. :(

Reply:
Yeah probably buggy.